plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1027,14 +1047,22 @@ private Optional<ConnectorTableLayout> getWriteLayout(Schema tableSchema, Partit
         return Optional.of(new ConnectorTableLayout(partitioningHandle, partitioningColumnNames, true));
     }
 
+    private static int getRootFieldId(Map<Integer, Integer> indexParents, int fieldId)
+    {
+        Integer sourceId = indexParents.get(fieldId);
```

Comment:
Retrieving the root field id can be done also iteratively by keeping the same conciseness